File: java/src/main/java/it/giannibombelli/wsc2026/payment/application/query/PaymentSummary.java
package it.giannibombelli.wsc2026.payment.application.query;

import it.giannibombelli.wsc2026.common.domain.primitive.ClientReference;
import it.giannibombelli.wsc2026.common.domain.primitive.Money;
import it.giannibombelli.wsc2026.common.domain.primitive.Timestamp;
import it.giannibombelli.wsc2026.payment.domain.payment.PaymentStatus;

import java.util.UUID;

public record PaymentSummary(
    UUID id,
    ClientReference clientReference,
    Money amount,
    PaymentStatus status,
    Timestamp requestedAt
) {
}

File: typescript/src/payment/application/query/paymentSummary.ts
// Read model riassuntivo di un Payment.

import type { ClientReference } from '@/common/domain/primitive/clientReference.js';
import type { Money } from '@/common/domain/primitive/money.js';
import type { Timestamp } from '@/common/domain/primitive/timestamp.js';
import type { PaymentStatus } from '../../domain/payment/paymentStatus.js';

export type PaymentSummary = {
  readonly id: string;
  readonly clientReference: ClientReference;
  readonly amount: Money;
  readonly status: PaymentStatus;
  readonly requestedAt: Timestamp;
};
